python/taichi/lang/kernel_arguments.py

```diff
@@ -57,8 +58,25 @@ def __init__(self, element_shape=(), layout=Layout.AOS):
         self.layout = layout
 
     def extract(self, x):
-        shape = tuple(x.shape)
+        from taichi.lang.matrix import MatrixNdarray, VectorNdarray
```

Comment:
move to top? (or will it cause circular import?)

Reply:
It will cause circular import :-(